File: AdventOfCode/2022/scripts/download.js

async function downloadData(year, day, session) {
    const url = `https://adventofcode.com/${year}/day/${day}/input`;
    console.log(url)

    const response = await fetch(url, {
        headers: {
            Cookie: `session=${session}`
        }
    });

    const data = response.text();
    return data;
}


import process from 'process'
import fs from 'fs';

async function run() {
    const parameters = process.argv.slice(2)
    const [name, year, day, session] = parameters;

    if (year === "" || day === "" || session==="") {
        console.log("usage: year day session");
        return;
    }

    console.log(`
    year: ${year}
    day: ${day}
    session: ${session}
    `);

    const dayN = Number.parseInt(day).toString()
    const data = downloadData(year, dayN, session);

    const fileName = `${dayN.padStart(2, '0')}.data.txt`;

    console.log(fileName);
    fs.writeFileSync(fileName, data);
}

run();

File: AdventOfCode/2021/download.py
import urllib.request

def download_data(year, day, session):
    url = r'''https://adventofcode.com/{}/day/{}/input'''.format(year, day)
    print(url)

    opener = urllib.request.build_opener()
    opener.addheaders = [('cookie', 'session={}'.format(session))]
    response = opener.open(url)

    #print(response)

    data = response.read().decode('utf-8')

    #print(data)
    return data


def write_file(file, data):
    f = open(file, 'w')
    f.write(data)
    f.close()


import sys

def run():
    name, year, day, session = sys.argv

    if (year == None or day == None or session== None):
        print("usage: year day session")
        return

    print('''
    year: {}
    day: {}
    session: {}
    '''.format(year, day, session))

    data = download_data(year, str(int(day)), session)

    file = "{}.data.txt".format(str(int(day)).zfill(2))

    print()
    print(file)

    write_file(file, data)

run()
